cmd: copy edits for import-url ref. (#1494)
iterative/dvc.org: content/docs/command-reference/import-url.md

metadata about the imported data to enable DVC efficiently determining whether
the local copy is out of date.

> `dvc repro` doesn't check and/or update generated `.dvc` files. Use `dvc update`

Author: Just thought that the two would be more readable here given that their subjects are different.

Reviewer: No, these are separate sentences. Also, out of scope.

Reviewer: "and/or" should just be "or" though, please change that since we're looking at this already.

metadata about the imported data to enable DVC efficiently determining whether
the local copy is out of date.

> `dvc repro` doesn't check or update generated `.dvc` files, use `dvc update` on

Reviewer: ```suggestion
Note that `dvc repro` doesn't check or update import `.dvc` files, use `dvc update` on
```
Can you please apply this and format the paragraph? A bit out of scope but it relates to something else you added...

Reviewer: Also, in line 48 can you change "In such a `.dvc`" for "In such an import `.dvc`"? Thanks

  running. DVC automatically expands this URL into a regular S3, SSH, GS, etc
  URL by appending `/path/to/file` to the `myremote`'s configured base path.

The `deps` field in the generated `.dvc` file contains a hash value along with
the `path` to track changes in the remote file/directory. The hash type depends
on the type of remote location (protocol) being tracked. See
[`dvc.lock` file](/doc/user-guide/dvc-files-and-directories#dvclock-file) for
info on hash types for specific remote location types.

> 

Reviewer: Yeah this whole paragraph is probably unnecessary but thanks. The ref. is already too long TBH. Users can find out about .dvc file specs in its dedicated guide, that's not the main purpose for a command ref.

Author: Okay, so where should I link `dvc.lock file` ?

Reviewer: Why did you decide to edit this command reference if there's no existing places that relate to dvc.lock? I'm going to just merge this with the few typo fixed but I'll have to uncheck that box in the original index so this can be re-evaluated.

Author: I actually thought that we needed to add some reference to the hash values in each file or describe where it was needed based on the original issue. But does this mean that there is no place we can add a reference to `dvc.lock` or provide info about different types of file hashes in `import-url`?

Reviewer: > does this mean that there is no place we can add a reference to dvc.lock or provide info about different types of file hashes in import-url?

Idk, can you answer that question? 🙂 Let's go back to analyzing the checkboxes of #1469 and please comment there if any one seems unaplicable.

  running. DVC automatically expands this URL into a regular S3, SSH, GS, etc
  URL by appending `/path/to/file` to the `myremote`'s configured base path.

The `deps` field in the generated `.dvc` file contains a hash value along with
the `path` to track changes in the remote file/directory. The hash type depends
on the type of remote location (protocol) being tracked. For instance,

> files/directories tracked from HTTP(s)/S3/Azure/GS remote locations will have an

Reviewer 2: I made a mistake with the comment. GS uses `md5`. Sorry about that.

Author: Okay, I'll change it in the DVC Files and Directories page in the next commit :v:

Reviewer: No worries @skshetry, glad you caught it though! Thanks for the followup. I updated #1469 and fixed the dvc.lock doc in another PR. But this brings some questions... I'll ping you somewhere else.

@utkarshsingh99 please update the paragraph above accordingly...

[ETag](https://en.wikipedia.org/wiki/HTTP_ETag) value in the `deps` field, while
files/directories tracked from SSH/local locations will have an `md5` hash
value. Files from HDFS remote locations use a `checksum` key in the `deps`
field.

> 

Reviewer: Ah thanks! But the intention of the ticket wasn't to repeat all this info here! Just double check that any existing mentions of hash values are correct. We should link "`.dvc` files" to the corresponding user guide though, so that people can see the full spec there, including types of hash values.

Reviewer: Also in line 111 can you capitalize the first word? "It". Thanks

Author: I see this link containing the info we need but how can I link precisely that paragraph which talks about the different hash values instead of the section `dvc.lock file`?
https://dvc.org/doc/user-guide/dvc-files-and-directories#dvclock-file

It would be great if we could link this paragraph:
> Regular dependencies and all kinds of outputs (including metrics and plots files) are also listed (per stage) in dvc.lock, but with an additional field to store the hash value of each file or directory tracked by DVC: md5 for local file system dependencies and SSH external dependencies, etag for HTTP, S3, Azure, and Google Cloud, and checksum for HDFS.

Reviewer: Can't link to a paragraph haha. That's not necessary, the anchor is enough... People have eyes 😉

Reviewer: This p is still there... Updated https://github.com/iterative/dvc.org/pull/1494#pullrequestreview-442823296. Resolving here.

  running. DVC automatically expands this URL into a regular S3, SSH, GS, etc
  URL by appending `/path/to/file` to the `myremote`'s configured base path.

See [`dvc.lock` file](/doc/user-guide/dvc-files-and-directories#dvclock-file)
for info on hash types for specific remote location types.

> 

Reviewer: Again, this ref. is already too long. Let's not add this.

Reviewer: I'll fix it.
```suggestion
```

```

`dvc import-url` generates an import stage
[`.dvc` file](/doc/user-guide/dvc-files-and-directories#dvc-files), and

> `dvc run` generates a regular stage (in

Author: Readability improvement. I think it can be changed back to original too, was just a potential enhancement.
Or we could also add 
```
..., while `dvc run` generates...
```

Reviewer: Out of scope. Please roll back.

I would decide now since you already noticed this paragraph but we are going to get rid of the "import stage" term soon so this whole sentence will have to be rewritten anyway.

Reviewer: BTW, appreciate the initiative! It's just that right now we want to have very small and focused PRs during this part of the process 🙂

The `etag` field in the `.dvc` file contains the
[ETag](https://en.wikipedia.org/wiki/HTTP_ETag) recorded from the HTTP request.
If the remote file changes, its ETag will be different. This metadata allows DVC

> to determine whether it is necessary to download it again.

Reviewer: Thanks. Let's do
```suggestion
to determine whether it's necessary to download it again.
```

The `etag` field in the `.dvc` file contains the
[ETag](https://en.wikipedia.org/wiki/HTTP_ETag) recorded from the HTTP request.
If the remote file changes, its ETag will be different. This metadata allows DVC
to determine whether its necessary to download it again.
to determine whether it's necessary to download it again.

> See [`.dvc` files](/doc/user-guide/dvc-files-and-directories#dvc-files) for
> more details on the format above.

You may want to get out of and remove the `example-get-started/` directory after
trying this example (especially if trying out the following one).

> 

Author: I believe this statement is irrelevant here since anyone could arrive at this page without going through the `User Guide/Getting-Started` and it does not hold a direct info about the `import-url` command. It seems like the content was borrowed from one of those pages.

Reviewer: I don't think you followed the entire examples section first. That would need to be done to determine this. Feel free to do so and submit a different PR if you find this unnecessary, but it's out of scope for this PR. Please roll back here. Thanks

Author: Alright.
But do you think it should be removed? 
> I don't think you followed the entire examples section first.  

Actually, since this is in `command reference`, I thought it would be good to keep this file independent of the `Getting Started` module

Reviewer: I don't think it should be removed. But like I mentioned, I would need to follow the entire examples section first to decide.

> would be good to keep this file independent of the Getting Started

It is. Again, I don't think you read the whole examples section 🙂 

Please roll back.

Author: Oh okay. Now I get it. It's just related to the example above.
My apologies :smile:
